src/Microsoft.PowerShell.Commands.Utility/commands/utility/MarkdownOptionCommands.cs

```diff
@@ -270,28 +257,30 @@ public class GetMarkdownOptionCommand : PSCmdlet
         /// </summary>
         protected override void EndProcessing()
         {
-            var option = this.CommandInfo.Module == null
-                // If we don't have a moduleInfo, like in PowerShell hosting scenarios, use a concurrent dictionary.
-                ? PSMarkdownOptionInfoCache.Get(Runspace.DefaultRunspace.InstanceId)
-                // If we have the moduleInfo then store are module scope variable
-                : this.CommandInfo.Module.SessionState.PSVariable.GetValue(MarkdownOptionInfoVariableName, new PSMarkdownOptionInfo());
-
-            WriteObject(option);
+            WriteObject(PSMarkdownOptionInfoCache.Get(this.CommandInfo));
         }
     }
 
     internal static class PSMarkdownOptionInfoCache
     {
         private static ConcurrentDictionary<Guid, PSMarkdownOptionInfo> markdownOptionInfoCache;
+        private const string MarkdownOptionInfoVariableName = "PSMarkdownOptionInfo";
 
         static PSMarkdownOptionInfoCache()
         {
             markdownOptionInfoCache = new ConcurrentDictionary<Guid, PSMarkdownOptionInfo>();
         }
 
-        internal static PSMarkdownOptionInfo Get(Guid runspaceId)
+        internal static PSMarkdownOptionInfo Get(CommandInfo command)
         {
-            if (markdownOptionInfoCache.TryGetValue(runspaceId, out PSMarkdownOptionInfo cachedOption))
+            // If we have the moduleInfo then store are module scope variable
+            if (command.Module != null)
+            {
+                return command.Module.SessionState.PSVariable.GetValue(MarkdownOptionInfoVariableName, new PSMarkdownOptionInfo()) as PSMarkdownOptionInfo;
```

Comment:
I wonder that the API to force creating default value before trying to get a value.